### ISSUE TITLE
roslaunch doesn't allows ssh authentication using empty password

### ISSUE THREAD

**Issue by 5tan** [ros/ros_comm#1827]
If we try following machine definition:
```
<machine
  name="mmm"
  address="10.20.73.2"
  user="root"
  password=""
  timeout="10"
  env-loader="/path/to/remote-env.sh"
/>
```
roslaunch attempts to authenticate using key instead of empty password.

**Comment by 5tan** [comment]
Fix: https://github.com/ros/ros_comm/pull/1826

**Comment by dirk-thomas** [comment]
Addressed by #1826.
